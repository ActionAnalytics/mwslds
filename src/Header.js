import React from 'react'

import bclogo from 'bootstrap-theme/dist/images/17_gov3_bc_logo.svg'
import bclogoSmall from 'bootstrap-theme/dist/images/01_gov3_bc_symbol.svg'

// import 'bootstrap-theme/dist/js/bootstrap.bundle'
import 'bootstrap-theme/dist/css/bootstrap-theme.min.css'

export default function Header(props) {
  return (
    <header>
      <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
        <div className="container">
          <a className="navbar-brand" href="https://www2.gov.bc.ca">
            <img className="img-fluid d-none d-md-block" src={bclogo} width="152" height="55" alt="B.C. Government Logo" />
            <img className="img-fluid d-md-none" src={bclogoSmall} width="61" height="43" alt="B.C. Government Logo" />
          </a>
          <div className="navbar-brand">
            {props.title}
          </div>
          <button
            className="navbar-toggler collapsed"
            type="button"
            data-toggle="collapse"
            data-target="#navbarNavAltMarkup" // TODO: dont use an id here
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>
          <div className="navbar-collapse collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav">
              {props.items && props.items.map(link => (
                <a key={link.name} className="nav-item nav-link" href={link.href}>{link.name}</a>
              ))}
            </div>
          </div>
        </div>
      </nav>
    </header>
  )
}
