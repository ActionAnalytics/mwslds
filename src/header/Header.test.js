import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import { createSerializer } from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import Header from './Header'

configure({ adapter: new Adapter() })
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))

describe('Header', () => {
  it('renders', () => {
    const header = shallow(<Header title="headerless horseman" />)
    expect(header).toMatchSnapshot()
  })

  it('renders links correctly', () => {
    const header = mount((
      <Header>
        <a href="">earth</a>
        <a href="">sun</a>
        <a href="">moon</a>
        <a href="">mars</a>
      </Header>
    ))
    expect(header).toMatchSnapshot()
  })
})
