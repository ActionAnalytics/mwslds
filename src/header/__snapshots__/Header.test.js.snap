// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  id="header"
>
  <div
    className="navbar navbar-default navbar-fixed-top"
    id="header-main"
  >
    <div
      className="container"
    >
      <div
        className="row"
        id="header-main-row"
      >
        <div
          className="col-sm-3 col-md-2 col-lg-2 header-main-left"
        >
          <div
            id="logo"
          >
            <a
              href="https://www2.gov.bc.ca"
            >
              <img
                alt="B.C. Government Logo"
                src="gov3_bc_logo.png"
              />
            </a>
          </div>
          <div
            id="access"
          >
            <ul>
              <li
                aria-label="Keyboard Tab Skip"
              >
                <a
                  aria-label="Skip to main content"
                  href="#main-content-anchor"
                >
                  Skip to main content
                </a>
              </li>
              <li
                aria-label="Keyboard Tab Skip"
              >
                <a
                  aria-label="Skip to navigation"
                  href="#main-content-anchor"
                  onClick={[Function]}
                >
                  Skip to navigation
                </a>
              </li>
              <li
                aria-label="Keyboard Tab Skip"
              >
                <a
                  aria-label="Accessibility Statement"
                  href="http://gov.bc.ca/webaccessibility/"
                >
                  Accessibility Statement
                </a>
              </li>
            </ul>
          </div>
          <button
            aria-expanded="true"
            aria-label="Burger Navigation"
            className="navbar-toggle gov-button-custom collapsed"
            data-target="#navbar"
            data-toggle="collapse"
            type="button"
          >
            <img
              alt="Open Menu"
              src="menu-open-mobile.png"
            />
          </button>
        </div>
        <div
          className="col-sm-8 col-md-8 col-lg-6 hidden-xs"
        >
          <div
            className="bcgov-title"
          >
            <h1>
              headerless horseman
            </h1>
          </div>
        </div>
        <div
          className="collapse navbar-collapse"
          id="navbar"
        >
          <ul
            className="nav navbar-nav"
          />
        </div>
      </div>
    </div>
    <div
      className="navigationRibbon"
    >
      <div
        className="level2Navigation"
      >
        <div
          className="container"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Header renders links correctly 1`] = `
<div
  id="header"
>
  <div
    className="navbar navbar-default navbar-fixed-top"
    id="header-main"
  >
    <div
      className="container"
    >
      <div
        className="row"
        id="header-main-row"
      >
        <div
          className="col-sm-3 col-md-2 col-lg-2 header-main-left"
        >
          <div
            id="logo"
          >
            <a
              href="https://www2.gov.bc.ca"
            >
              <img
                alt="B.C. Government Logo"
                src="gov3_bc_logo.png"
              />
            </a>
          </div>
          <div
            id="access"
          >
            <ul>
              <li
                aria-label="Keyboard Tab Skip"
              >
                <a
                  aria-label="Skip to main content"
                  href="#main-content-anchor"
                >
                  Skip to main content
                </a>
              </li>
              <li
                aria-label="Keyboard Tab Skip"
              >
                <a
                  aria-label="Skip to navigation"
                  href="#main-content-anchor"
                  onClick={[Function]}
                >
                  Skip to navigation
                </a>
              </li>
              <li
                aria-label="Keyboard Tab Skip"
              >
                <a
                  aria-label="Accessibility Statement"
                  href="http://gov.bc.ca/webaccessibility/"
                >
                  Accessibility Statement
                </a>
              </li>
            </ul>
          </div>
          <button
            aria-expanded="true"
            aria-label="Burger Navigation"
            className="navbar-toggle gov-button-custom collapsed"
            data-target="#navbar"
            data-toggle="collapse"
            type="button"
          >
            <img
              alt="Open Menu"
              src="menu-open-mobile.png"
            />
          </button>
        </div>
        <div
          className="col-sm-8 col-md-8 col-lg-6 hidden-xs"
        >
          <div
            className="bcgov-title"
          >
            <h1 />
          </div>
        </div>
        <div
          className="collapse navbar-collapse"
          id="navbar"
        >
          <ul
            className="nav navbar-nav"
          >
            <li
              key="0"
            >
              <a
                href=""
              >
                earth
              </a>
            </li>
            <li
              key="1"
            >
              <a
                href=""
              >
                sun
              </a>
            </li>
            <li
              key="2"
            >
              <a
                href=""
              >
                moon
              </a>
            </li>
            <li
              key="3"
            >
              <a
                href=""
              >
                mars
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="navigationRibbon"
    >
      <div
        className="level2Navigation"
      >
        <div
          className="container"
        >
          <a
            href=""
          >
            earth
          </a>
          <a
            href=""
          >
            sun
          </a>
          <a
            href=""
          >
            moon
          </a>
          <a
            href=""
          >
            mars
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
